### ISSUE TITLE
add-hook does not allow overriding existing handler

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nhooks#9]
Currently calling `add-hook` over a handler with an existing name is a no-op.

But what if we want to replace the handler in question?

Right now, the only way is to remove the handler then re-add it.

I wonder if it's really what's most expected.  It seems to me that we would normally expect the latest version of a handler to be registered.

So I suggest reverting the behaviour: add-hook replace handlers with the same name, and if the user does not want it, they can just check for existence with `find-handler`.

Thoughts?  @BlueFlo0d

**Comment by BlueFlo0d** [comment]
Makes sense. I'm a bit busy to following days but will be able to do it this week.

**Comment by BlueFlo0d** [comment]
Some design detail: 
- Should `add-hook` push the handler to the front of the precedence list, or keep it in place?
- If a handler is disabled, should `add-hook` enable it?

**Comment by Ambrevar** [comment]
All good questions!

- I would place the handler first, because the user knows what was done
  last but does no necessarily remember an old handler position, so the
  user would expect the newly-added handler to be in the front.

- Following the same logic, the user may not remember the state of the
  old handler and expect the newly-added handler to be enabled like any
  other handler.

This also makes the code very simple!

That said, maybe we want to add an add-hook option to preserve position and state.

**Comment by BlueFlo0d** [comment]
> That said, maybe we want to add an add-hook option to preserve position and state.

I think this option is not really necessary. If the user want to do it in place, they can just redefine the symbol function (the most common use case), or replace the `fn` slot of `handler`.

That said we should probably update `find-handler`? It looks a bit out of sync with the rest. Once it is taken care of, users can then `(unless (find-handler ...) (add-hook ...))` to ensure a handler is present.
